beam/core/src/main/java/cz/o2/proxima/beam/direct/io/OffsetRestrictionTracker.java

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright 2017-2020 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.beam.direct.io;
+
+import com.google.common.base.Preconditions;
+import cz.o2.proxima.direct.commitlog.Offset;
+import java.io.Serializable;
+import java.util.Objects;
+import lombok.Getter;
+import lombok.ToString;
+import org.apache.beam.sdk.transforms.splittabledofn.HasDefaultTracker;
+import org.apache.beam.sdk.transforms.splittabledofn.RestrictionTracker;
+import org.apache.beam.sdk.transforms.splittabledofn.SplitResult;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * A {@link RestrictionTracker} for {@link Offset Offsets} read from {@link
+ * cz.o2.proxima.direct.commitlog.CommitLogReader}
+ */
+public class OffsetRestrictionTracker
+    extends RestrictionTracker<OffsetRestrictionTracker.StartEndOffset, Offset> {
+
+  @ToString
+  public static class StartEndOffset
```

Comment:
This class name was super confusing to me at the beginning. Can we rename it to something like `Range` as this RestrictionTracker basically tries to claim a particular "Offset" from a given range?

Reply:
❤️